tests/glsl-intrinsic/shader-subgroup/shader-subgroup-ballot.slang

```diff
@@ -0,0 +1,134 @@
+bool test1BroadcastX() {
+    return true
+        && subgroupBroadcast(T(1), 0) == T(1)
+        && subgroupBroadcastFirst(T(1)) == T(1)
+        ;
+}
+__generic<T : __BuiltinLogicalType, let N : int>
+bool testVBroadcastX() {
+    typealias gvec = vector<T, N>;
+
+    return true
+        && subgroupBroadcast(gvec(T(1)), 0) == gvec(T(1))
+        && subgroupBroadcastFirst(gvec(T(1))) == gvec(T(1))
+        ;
+}
+
+__generic<T : __BuiltinFloatingPointType>
+bool test1BroadcastX() {
+    return true
+        && subgroupBroadcast(T(1), 0) == T(1)
+        && subgroupBroadcastFirst(T(1)) == T(1)
+        ;
+}
+__generic<T : __BuiltinFloatingPointType, let N : int>
+bool testVBroadcastX() {
+    typealias gvec = vector<T, N>;
+
+    return true
+        && subgroupBroadcast(gvec(T(1)), 0) == gvec(T(1))
+        && subgroupBroadcastFirst(gvec(T(1))) == gvec(T(1))
+        ;
+}
+bool testBroadcastX() {
+    return true
+        && test1BroadcastX<float>()
+        && testVBroadcastX<float, 2>()
+        && testVBroadcastX<float, 3>()
+        && testVBroadcastX<float, 4>()
+        && test1BroadcastX<double>() // WARNING: intel GPU's lack FP64 support
+        && testVBroadcastX<double, 2>()
+        && testVBroadcastX<double, 3>()
+        && testVBroadcastX<double, 4>()
+        && test1BroadcastX<half>() 
+        && testVBroadcastX<half, 2>()
+        && testVBroadcastX<half, 3>()
+        && testVBroadcastX<half, 4>()
+        && test1BroadcastX<int>()
+        && testVBroadcastX<int, 2>()
+        && testVBroadcastX<int, 3>()
+        && testVBroadcastX<int, 4>()
+        && test1BroadcastX<int8_t>() 
+        && testVBroadcastX<int8_t, 2>()
+        && testVBroadcastX<int8_t, 3>()
+        && testVBroadcastX<int8_t, 4>()
+        && test1BroadcastX<int16_t>() 
+        && testVBroadcastX<int16_t, 2>()
+        && testVBroadcastX<int16_t, 3>()
+        && testVBroadcastX<int16_t, 4>()
+        && test1BroadcastX<int64_t>() 
+        && testVBroadcastX<int64_t, 2>()
+        && testVBroadcastX<int64_t, 3>()
+        && testVBroadcastX<int64_t, 4>()
+        && test1BroadcastX<uint>()
+        && testVBroadcastX<uint, 2>()
+        && testVBroadcastX<uint, 3>()
+        && testVBroadcastX<uint, 4>()
+        && test1BroadcastX<bool>()
+        && testVBroadcastX<bool, 2>()
+        && testVBroadcastX<bool, 3>()
+        && testVBroadcastX<bool, 4>()
+        ;
+}
+
+bool testBallot() {
+    return true 
+        && (subgroupBallot(true).x == 0xFFFFFFFF)
+        && (subgroupInverseBallot(uvec4(0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF)) == true)
+        && (subgroupBallotBitExtract(uvec4(0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF), 0) == true)
+        && (subgroupBallotBitCount(uvec4(0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF)) == 32)
+        && (subgroupBallotInclusiveBitCount(uvec4(0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF)) != 0)
+#ifdef TEST_when_glsl_subgroupBallotExclusiveBitCount_is_not_bugged
```

Comment:
Can you fix this?
It seems like the error message is following:
```
glsl.meta.slang(2841): warning 41000: unreachable code detected
        return;
        ^~~~~~
```
When the issue is reproduced, you can see the error message in a file, "shader-subgroup-ballot.slang.3.actual"

Reply:
I can't reproduce the issue despite anything I try, may be due to the new branch having changes